### DISCUSSION TITLE
Axios Instance 400 returned as Success

### DISCUSSION THREAD

**Discussion by waelbenmustapha** [axios/axios#5875] [Q&A]
### Describe the bug

when i use an axios instance i created , the 400 status code is returned in .then() as success but when i use the default axios it return as an error in the .catch() , i don't know why this is happening , i tried adding the validateStatus : () => return false and it keeps returning everything as success . 

### To Reproduce

```
import axios from "axios";

export default axios.create({
  baseURL: "http://192.168.1.45:3000",
  headers: {
    "Content-type": "application/json",
  },
});

```
i imported the axiosToken on project initialization and injected my token and made an interceptor for the 401 error status code
```
axiosTokenInstance.defaults.headers.common[
    "Authorization"
  ] = `Bearer ${auth.user.token}`;


  axiosTokenInstance.interceptors.response.use(
    (response) => {
      return response;
    },
    (error) => {
      if (error.response.status === 401) {
        auth.removeAsyncUser();
      }
      return error;
    }
  );
```

**Comment by philBrown** [comment]
The error is in your interceptor. Failed requests should return a rejected promise
```js
return Promise.reject(error);
```

See the docs ~ https://github.com/axios/axios#interceptors

**Comment by teobmg** [comment]
I think that is more related to the default value of the `validateStatus` option.
Creating a new axios instance, if you define it, you have your desired behavior.

**Comment by Mujaddadi** [comment]
Same problem.  The issue seems to be with the Axios treating 400s differently in the interceptor.  I try adding this but it still treats 400s in then and not in catches.

`axios.defaults.validateStatus = status => status >= 200 && status < 300;`

Maybe I am missing something.

**Comment by Mujaddadi** [comment]
I am doing it in a hacky way. In the success interceptor function, I throw an error

`    if (response.status >= 400) {
      throw response;
    }`
